Sources/Basic/TemporaryFile.swift

```diff
@@ -45,18 +29,52 @@ private extension TempFileError {
 /// - Returns: Path to directory in which temporary file should be created.
 public func determineTempDirectory(_ dir: AbsolutePath? = nil) throws -> AbsolutePath {
     // FIXME: Add other platform specific locations.
-    let tmpDir = dir ?? cachedTempDirectory
+    let tmpDir = dir ?? AbsolutePath(NSTemporaryDirectory())
     guard localFileSystem.isDirectory(tmpDir) else {
-        throw TempFileError.couldNotFindTmpDir
+        throw TempFileOrDirError.couldNotFindTmpDir
     }
     return tmpDir
 }
 
-/// Returns temporary directory location by searching relevant env variables.
-/// Evaluates once per execution.
-private var cachedTempDirectory: AbsolutePath = {
-    return AbsolutePath(Process.env["TMPDIR"] ?? Process.env["TEMP"] ?? Process.env["TMP"] ?? "/tmp/")
-}()
+let ALPHANUMERIC_CHARS = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+
+func createMkstempLikeName(prefix: String, suffix: String) -> String {
+    let randomMiddleStringLen = 6
+    let randomMiddleString =
+        String((0..<randomMiddleStringLen).map { _ in ALPHANUMERIC_CHARS.randomElement()! })
+    return prefix + "." + randomMiddleString + suffix
+}
+
+func mkstemps(dir: AbsolutePath, prefix: String, suffix: String) throws -> AbsolutePath {
```

Comment:
I am a little nervous about this. I think we should at least use mkstemps where it's available.

Reply:
I think that the logic is quite straightforward.
Why are you nervous about it? We can discuss it

Reply:
My team changed my mind, i'll implement this in a posix way and additionally a windows specific way